### ISSUE TITLE
The solana-web3.js confirmTransaction function with "confirmed" is stuck when transaction had been finalized

### ISSUE THREAD

**Issue by wngasinur** [solana-labs/solana-web3.js#2713]
#### Problem

I'm getting random issue while calling `confirmTransaction`

```
// send transaction with "confirmed" blockhash

// transaction seems to be finalized before below line is being executed

connection.confirmTransaction(
{
    blockhash: latestBlockValue.blockhash,
    lastValidBlockHeight: latestBlockValue.lastValidBlockHeight,
    signature: finalSignature,
},  "confirmed")
```

The confirmTransaction never returns and it is throwing block height expiry exception. Is this expected behaviour ?

**Comment by buffalojoec** [comment]
> send transaction with "confirmed" blockhash

Could you elaborate on what you mean by this?

**Comment by steveluscher** [comment]
This method _only_ ever was designed to confirm _recent_ transactions; that is to say transactions for which the signature status is still in the status cache.

Your confusion here is exactly the reason why I renamed this method to `confirmRecentTransaction()` in the new web3.js.

https://github.com/solana-labs/solana-web3.js/blob/master/packages/transaction-confirmation/src/confirmation-strategy-recent-signature.ts

tl;dr if you have reason to believe that a signature has fallen out of the signature cache, because it's old, you're looking for the `getSignatureStatuses` RPC method with the `searchTransactionHistory` param set to `true`.

**Comment by wngasinur** [comment]
Can you explain what do you mean by status still in cache? And how do I know if the status still in cache? My ask is just to poll the signature status once the transaction is submitted.

**Comment by wngasinur** [comment]
@steveluscher i have noticed there was a similar closed bug https://github.com/solana-labs/solana-web3.js/issues/1107 , can you confirm if this resolved? Cause I am getting the same problem as described there
